tests/unit/platform_apis_test.c

```diff
@@ -45,13 +63,39 @@ CTEST2(platform_api, test_platform_buffer_init)
    platform_status rc = STATUS_NO_MEMORY;
 
    buffer_handle bh;
+   ZERO_CONTENTS(&bh);
 
    rc = platform_buffer_init(&bh, KiB);
    ASSERT_TRUE(SUCCESS(rc));
    ASSERT_TRUE(bh.addr != NULL);
    ASSERT_TRUE(bh.addr == platform_buffer_getaddr(&bh));
+   ASSERT_TRUE(bh.length == KiB);
 
    rc = platform_buffer_deinit(&bh);
    ASSERT_TRUE(SUCCESS(rc));
    ASSERT_TRUE(bh.addr == NULL);
+   ASSERT_TRUE(bh.length == 0);
+}
+
+/*
+ * Test failure to mmap() a very large buffer_init().
+ */
+CTEST2(platform_api, test_platform_buffer_init_fails_for_very_large_length)
+{
+   platform_status rc = STATUS_NO_MEMORY;
+
+   buffer_handle bh;
+   ZERO_CONTENTS(&bh);
+
+   size_t length = (1024 * KiB * GiB);
+
+   // This is expected to fail as length is very large
+   rc = platform_buffer_init(&bh, length);
```

Comment:
From code inspection and manual testing, the length set on L90 is sufficiently big-enough to cause `mmap()` to fail w/ an error.

Should this approach be tightened further? (I think this is sufficient.)